dpdata/plugins/ase.py

```diff
@@ -187,6 +188,135 @@ def to_labeled_system(self, data, *args, **kwargs):
+        end: Optional[int] = None,
+        step: Optional[int] = 1,
+        **kwargs,
+    ) -> dict:
+        """Read ASE's trajectory file to `System` of multiple frames.
+
+        Parameters
+        ----------
+        file_name : str
+            ASE's trajectory file
+        begin : int, optional
+            begin frame index
+        end : int, optional
+            end frame index
+        step : int, optional
+            frame index step
+        **kwargs : dict
+            other parameters
+
+        Returns
+        -------
+        dict_frames: dict
+            a dictionary containing data of multiple frames
+        """
+        traj = Trajectory(file_name)
+        sub_traj = traj[begin:end:step]
+        dict_frames = {
+            "atom_names": None,
+            "atom_numbs": None,
+            "atom_types": None,
+            "orig": None,
+            "nopbc": None,
+            "cells": [None] * len(sub_traj),
+            "coords": [None] * len(sub_traj),
+        }
+        for i, atoms in enumerate(sub_traj):
+            tmp = ASEStructureFormat().from_system(atoms)
+            if i == 0:
+                dict_frames["atom_names"] = tmp["atom_names"]
+                dict_frames["atom_numbs"] = tmp["atom_numbs"]
+                dict_frames["atom_types"] = tmp["atom_types"]
+                dict_frames["orig"] = tmp["orig"]
+                dict_frames["nopbc"] = tmp["nopbc"]
+            dict_frames["cells"][i] = tmp["cells"][0]
+            dict_frames["coords"][i] = tmp["coords"][0]
+
+        ## Convert to numpy array
+        dict_frames["cells"] = np.asarray(dict_frames["cells"])
+        dict_frames["coords"] = np.asarray(dict_frames["coords"])
+
+        return dict_frames
+
+    def from_labeled_system(
+        self,
+        file_name: str,
+        begin: Optional[int] = 0,
+        end: Optional[int] = None,
+        step: Optional[int] = 1,
+        **kwargs,
+    ) -> dict:
+        """Read ASE's trajectory file to `System` of multiple frames.
+
+        Parameters
+        ----------
+        file_name : str
+            ASE's trajectory file
+        begin : int, optional
+            begin frame index
+        end : int, optional
+            end frame index
+        step : int, optional
+            frame index step
+        **kwargs : dict
+            other parameters
+
+        Returns
+        -------
+        dict_frames: dict
+            a dictionary containing data of multiple frames
+        """
+        traj = Trajectory(file_name)
+        sub_traj = traj[begin:end:step]
+        dict_frames = {
+            "atom_names": None,
+            "atom_numbs": None,
+            "atom_types": None,
+            "orig": None,
+            "nopbc": None,
+            "cells": [None] * len(sub_traj),
+            "coords": [None] * len(sub_traj),
+            "energies": [None] * len(sub_traj),
+            "forces": [None] * len(sub_traj),
+            "virials": [None] * len(sub_traj),
```

Comment:
This might cause errors (not sure). Need a unit test for the virial not existing case.

Reply:
I put a new unlabeled traj file for this test

Reply:
why error?

Reply:
I think lines 305 to 310  can prevent this